### ISSUE TITLE
Transition matrix not recomputed for new parameters

### ISSUE THREAD

**Issue by joshuapark00** [theislab/cellrank#575]
Hello, I'm enjoying using the package you've developed--it's very well documented and easy to use! I ran into one small issue while using it: when I try to compute initial/terminal states with a new connectivity weight, it ignores the new weight and uses any precomputed matrix (i.e. matrix using old connectivity weight) to do further calculations. 

The line I'm running is:
cr.tl.terminal_states(adata, cluster_key='seurat_clusters', weight_connectivities=0.6)

I can avoid this mainly by reloading the anndata without any stored transition matrix, but it would be great if we could try different parameter settings without reloading it each time. Thanks!

**Comment by michalk8** [comment]
Hi @joshuapark00 
currently, we don't store any info in `AnnData` which could determine whether connectivity weight has changed.
Usually, one could run e.g.:
```python
cr.tl.transition_matrix(adata, weight_connectivities=0.6)
cr.tl.terminal_states(adata, cluster_key='seurat_clusters')
```
but I can see the convenience of just using 1 function. In #557, I've added a new parameter `force_recompute: bool = False` so that running:
```python
 cr.tl.terminal_states(adata, cluster_key='seurat_clusters', weight_connectivities=0.6, force_recompute=True)
```
will always recompute the transition matrix, even if one is present. This is supposed to be a middle ground for now.

**Comment by michalk8** [comment]
It's on `dev` branch, closed via #577 .
